app/controllers/fhir/r4/api_controller.rb

```diff
@@ -0,0 +1,493 @@
+
+    # Parse in the FHIR::Patient
+    contents = FHIR.from_contents(request.body.string)
+    status_bad_request && return if contents.nil? || !contents.valid?
+
+    resource_type = params.permit(:resource_type)[:resource_type]&.downcase
+    case resource_type
+    when 'patient'
+      updates = Patient.from_fhir(contents).select { |_k, v| v.present? }
+      resource = get_patient(params.permit(:id)[:id])
+    else
+      status_bad_request && return
+    end
+
+    status_forbidden && return if resource.nil?
+
+    # Try to update the resource
+    status_bad_request && return if updates.nil? || !resource.update(updates)
+
+    status_ok(resource.as_fhir) && return
+  rescue StandardError
+    render json: operation_outcome_fatal.to_json, status: :internal_server_error
+  end
+
+  # Create a resource given a type.
+  #
+  # Supports (writing): Patient
+  #
+  # POST /fhir/r4/[:resource_type]
+  def create
+    status_unsupported_media_type && return unless content_type_header?
+
+    # Parse in the FHIR::Patient
+    contents = FHIR.from_contents(request.body.string)
+    status_bad_request && return if contents.nil? || !contents.valid?
+
+    resource_type = params.permit(:resource_type)[:resource_type]&.downcase
+    case resource_type
+    when 'patient'
+      # Construct a Sara Alert Patient
+      resource = Patient.new(Patient.from_fhir(contents))
+
+      # Responder is self
+      resource.responder = resource
+
+      # Creator is authenticated user
+      resource.creator = current_resource_owner
+
+      # Jurisdiction is the authenticated user's jurisdiction
+      resource.jurisdiction = current_resource_owner.jurisdiction
+
+      # Generate a submission token for the new monitoree
+      resource.submission_token = SecureRandom.hex(20) # 160 bits
+    end
+
+    status_bad_request && return if resource.nil?
+
+    status_bad_request && return unless resource.save
+
+    if resource_type == 'patient'
+      # Send enrollment notification
+      resource.send_enrollment_notification
+
+      # Create a history for the enrollment
+      history = History.new
+      history.created_by = current_resource_owner.email
+      history.comment = 'User enrolled monitoree via API.'
+      history.patient = resource
+      history.history_type = 'Enrollment'
+      history.save
+    end
+    status_created(resource.as_fhir) && return
+  rescue StandardError
+    render json: operation_outcome_fatal.to_json, status: :internal_server_error
+  end
+
+  # Return a FHIR Bundle containing results that match the given query.
+  #
+  # Supports (searching): Patient, Observation, QuestionnaireResponse
+  #
+  # GET /fhir/r4/[:resource_type]?parameter(s)
+  def search
+    status_not_acceptable && return unless accept_header?
+
+    resource_type = params.permit(:resource_type)[:resource_type]&.downcase
+    search_params = params.slice('family', 'given', 'telecom', 'email', 'subject', 'active',
+                                 '_count', '_id')
```

Comment:
Unless there is some magic ruby gem doing these, we should scrub these for SQL, e.g.

`&family="George'; drop database disease_trakker;--"`

https://imgs.xkcd.com/comics/exploits_of_a_mom.png

Reply:
Yeah, the `where`s below do escape, since we're not just passing one big interpolated string, but instead using the `?` and then param.